networkx/generators/geometric.py

```diff
@@ -844,3 +845,205 @@ def thresholded_random_geometric_graph(
+    and Marián Boguná. "Hyperbolic geometry of complex networks."
+    Physical Review E 82, no. 3 (2010): 036106.
+
+    """
+    if beta <= 0:
+        raise nx.NetworkXError("The parameter beta cannot be smaller or equal to 0.")
+
+    if kappas is not None:
+        if not all((n is None, gamma is None, mean_degree is None)):
+            raise nx.NetworkXError(
+                "When kappas is input, n, gamma and mean_degree must not be."
+            )
+
+        n = len(kappas)
+        mean_degree = sum(kappas) / len(kappas)
+    else:
+        if any((n is None, gamma is None, mean_degree is None)):
+            raise nx.NetworkXError(
+                "Please provide either kappas, or all 3 parameters: n, gamma and mean_degree."
+            )
+
+        # Generate hidden degrees from the powerlaw distribution with given exponent `gamma`
+        #  and mean value `mean_degree`
+        kappa_0 = (
+            (1 - 1 / n)
+            / (1 - n ** ((2 - gamma) / (gamma - 1)))
+            * (gamma - 2)
+            / (gamma - 1)
+            * mean_degree
+        )
+        kappa_c = kappa_0 * n ** (1 / (gamma - 1))
+
+        kappas = {}
+        for i in range(n):
+            kappas[i] = kappa_0 * (
+                1 - seed.uniform(0, 1) * (1 - (kappa_c / kappa_0) ** (1 - gamma))
+            ) ** (1 / (1 - gamma))
+
+    G = nx.empty_graph(kappas.keys())
+    R = n / (2 * math.pi)
+
+    # Approximate values for mu in the thermodynamic limit (when n -> infinity)
+    if beta > 1:
+        mu = beta * math.sin(math.pi / beta) / (2 * math.pi * mean_degree)
+    elif beta == 1:
+        mu = 1 / (2 * mean_degree * math.log(n))
+    else:
+        mu = (1 - beta) / (2**beta * mean_degree * n ** (1 - beta))
+
+    # Generate random positions on a circle
+    thetas = {k: seed.uniform(0, 2 * math.pi) for k in kappas}
+    mapping = dict(zip(range(n), kappas.keys()))
+
+    for i in range(n):
+        for j in range(i):
+            k, l = mapping[i], mapping[j]
+            angle = math.pi - math.fabs(math.pi - math.fabs(thetas[k] - thetas[l]))
+            dij = math.pow(R * angle, beta)
+            mu_kappas = math.pow(mu * kappas[k] * kappas[l], max(1, beta))
+            p_ij = 1 / (1 + dij / mu_kappas)
+
+            # Create an edge with a certain connection probability
+            if seed.random() < p_ij:
+                G.add_edge(k, l)
+
+    nx.set_node_attributes(G, thetas, "theta")
+    nx.set_node_attributes(G, kappas, "kappa")
+
+    # Map hidden degrees into the radial coordiantes
+    radii = {}
+    zeta = 1 if beta > 1 else 1 / beta
+    kappa_0 = min(list(kappas.values()))
+    R_hat = (2 / zeta) * math.log(n / math.pi) - (2 * max(1, beta)) / (
+        beta * zeta
+    ) * math.log(mu * kappa_0**2)
+
+    for k, kappa in kappas.items():
+        radii[k] = R_hat - (2 * max(1, beta)) / (beta * zeta) * math.log(
+            kappa / kappa_0
+        )
```

Comment:
Here's an attempt to improve readability (I used R_c for curvature, but you may have a better name):
```python
    R_c = 2 * max(1, beta) / (beta * zeta)
    R_hat = (2 / zeta) * math.log(n / math.pi) - R_c * math.log(mu * kappa_0**2)
    R_max = R_hat + R_c * math.log(kappa_0)

    for node, kappa in kappas.items():
        radii[node] = R_max - R_c * math.log(kappa)
```

You could also replace the loop with a dict-comprehension like:
```python
    radii = {node: R_max - R_c * math.log(kappa) for node, kappa in kappas.items()}
```

Reply:
Yeah, I would go with removing the square term and simplifying a bit.